src/app/tests/TestReadInteraction.cpp

```diff
@@ -1710,12 +1709,13 @@ void TestReadInteraction::TestSubscribeUrgentWildcardEvent(nlTestSuite * apSuite
         NL_TEST_ASSERT(apSuite, engine->GetNumActiveReadHandlers(ReadHandler::InteractionType::Subscribe) == 1);
 
         GenerateEvents(apSuite, apContext);
-        NL_TEST_ASSERT(apSuite, !delegate.mpReadHandler->mFlags.Has(ReadHandler::ReadHandlerFlags::HoldReport));
+        NL_TEST_ASSERT(apSuite, delegate.mpReadHandler->mFlags.Has(ReadHandler::ReadHandlerFlags::HoldReport));
         NL_TEST_ASSERT(apSuite, delegate.mpReadHandler->IsDirty() == true);
         delegate.mGotEventResponse = false;
         delegate.mGotReport        = false;
-        ctx.DrainAndServiceIO();
-        NL_TEST_ASSERT(apSuite, delegate.mGotEventResponse);
+        // wait for min interval 2 seconds, in test, we pick up 2.1 seconds, then the urgent event would be sent out
+        ctx.GetIOContext().DriveIOUntil(System::Clock::Milliseconds32(2100), [&]() { return delegate.mGotEventResponse == true; });
```

Comment:
This doesn't actually validate that the event _doesn't get_ sent in the min interval right?

Reply:
thanks for pointing out the issue, just add additional check to ensure no event is received within 2 seconds(in test, we use 1.9 seconds considering the potential time variation), then wait for 0.5 seconds, and validate that event is received.